File: Backend/src/main/java/com/ssafy/pennypal/domain/card/entity/Category.java
package com.ssafy.pennypal.domain.card.entity;


import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "category_id")
    Long categoryId;

    @Column(name = "category_shopping")
    Double categoryShopping;

    @Column(name = "category_culture")
    Double categoryCulture;

    @Column(name = "category_transportation")
    Double categoryTransportation;

    @Column(name = "category_car")
    Double categoryCar;

    @Column(name = "category_food")
    Double categoryFood;

    @Column(name = "category_education")
    Double categoryEducation;

    @Column(name = "category_housing_communication")
    Double categoryHousingCommunication;

    @Column(name = "category_travel")
    Double categoryTravel;

    @Column(name = "category_medical")
    Double categoryMedical;

    @Column(name = "category_financial_insurance")
    Double categoryFinancialInsurance;

    @Column(name = "category_others")
    Double categoryOthers;

    @OneToOne(mappedBy = "category")
    private Card card;
}

File: Frontend/src/pages/expenditure/model/category.ts
import { MemberCategoryPercentage } from '@/pages/expenditure/index';

export function isMemberCategoryPercentage(obj: any): obj is MemberCategoryPercentage {
    return (
        'category_car' in obj &&
        'category_culture' in obj &&
        'category_education' in obj &&
        // 모든 필수 카테고리 키들에 대해 확인
        'category_travel' in obj
    );
}

export const categoryLabels = {
    category_car: '자동차',
    category_culture: '문화',
    category_education: '교육',
    category_financial_insurance: '금융/보험',
    category_food: '식비',
    category_housing_communication: '주거/통신',
    category_medical: '의료',
    category_others: '기타',
    category_shopping: '쇼핑',
    category_transportation: '교통',
    category_travel: '항공/여행',
};

export type Props = {
    setReady: any;
};
